implementation/contracts/DepositLog.sol

```diff
@@ -99,15 +105,26 @@ contract DepositLog {
     //
 
     /// @notice             Checks if an address is an allowed logger
-    /// @dev                Calls the system to check if the caller is a Deposit
+    /// @dev                checks tbtcDepositToken to see if the caller represents
+    ///                     an existing deposit.
     ///                     We don't require this, so deposits are not bricked if the system borks
     /// @param  _caller     The address of the calling contract
     /// @return             True if approved, otherwise false
     /* solium-disable-next-line no-empty-blocks */
-    function approvedToLog(address _caller) public pure returns (bool) {
-        /* TODO: auth via system */
-        _caller;
-        return true;
+    function approvedToLog(address _caller) public view returns (bool) {
+        return tbtcDepositToken.exists(uint256(_caller));
+    }
+
+    /// @notice               Sets the tbtcDepositToken contract.
+    /// @dev                  The contract is used by `approvedToLog` to check if the
+    ///                       caller is a Deposit contract. This should only be called once.
+    /// @param  _tbtcDepositTokenAddress  The address of the tbtcDepositToken.
+    function setTbtcDepositToken(address _tbtcDepositTokenAddress) public {
```

Comment:
this should be internal - otherwise anyone could call this before `TBTCSystem.initialize` is called. 
it is not very elegant though that this is needed in the first place, see comment on shadowed variable - the token is also available in the system contract.

Reply:
> this should be internal

indeed >.<

> it is not very elegant though that this is needed in the first place

Agree here, There's definitely a cleaner way to do this